app/models/katello/content_view_puppet_environment.rb

```diff
@@ -34,8 +34,12 @@ def content_type
       Repository::PUPPET_TYPE
     end
 
-    def puppet?
-      true
+    def unprotected
+      false
+    end
+
+    def checksum_type
+      nil
     end
```

Comment:
we really need to turn unprotected into an attribute on the content view and look it up, but thats a feature we should add at some point.  

as for Checksum_type, you may add a check to anything that calls checksum_type to only call it for yum repos